packages/ember-data/lib/system/model/model.js

```diff
@@ -28,6 +29,15 @@ var retrieveFromCurrentState = Ember.computed('currentState', function(key) {
   @uses Ember.Evented
 */
 var Model = Ember.Object.extend(Ember.Evented, {
+
+  init: function() {
+    this._super(...arguments);
+
+    // caches for the relationship references
+    this.belongsToReferences = {};
+    this.hasManyReferences = {};
+  },
```

Comment:
Not sure actually, keep them on internal model for now
On Jun 11, 2015 4:44 AM, "Igor Terzic" <terzicigor@gmail.com> wrote:

> No
> On Jun 11, 2015 4:41 AM, "Clemens Müller" <notifications@github.com>
> wrote:
>
>> In packages/ember-data/lib/system/model/model.js
>> <https://github.com/emberjs/data/pull/3303#discussion_r32212306>:
>>
>> > @@ -28,6 +29,15 @@ var retrieveFromCurrentState = Ember.computed('currentState', function(key) {
>> >    @uses Ember.Evented
>> >  */
>> >  var Model = Ember.Object.extend(Ember.Evented, {
>> > +
>> > +  init: function() {
>> > +    this._super(...arguments);
>> > +
>> > +    // caches for the relationship references
>> > +    this.belongsToReferences = {};
>> > +    this.hasManyReferences = {};
>> > +  },
>>
>> Should the references be stored on the relationship states?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/emberjs/data/pull/3303/files#r32212306>.
>>
>